src/prefect/tasks/prefect/flow_run.py

```diff
@@ -106,4 +113,23 @@ def run(
 
         # grab the ID for the most recent version
         flow_id = flow[0].id
-        return client.create_flow_run(flow_id=flow_id, parameters=parameters)
+
+        # providing an idempotency key ensures that retries for this task
+        # will not create additional flow runs
+        flow_run_id = client.create_flow_run(
+            flow_id=flow_id,
+            parameters=parameters,
+            idempotency_key=context.get("flow_run_id"),
+        )
```

Comment:
Not sure if I'm overthinking, and feel free to ignore, but what if I do want my flow to kick off the run again on retries?

Reply:
Perhaps the `task_run_count` should be incorporated as part of the `idempotency_key`, and failures in the network request handled internal to the task instead of external with the retry mechanism. So retries for this task would be retrying the whole flow, not retrying the request to start the flow.

```suggestion
        # Providing an idempotency key ensures that retries for this network request
        # will not create additional flow runs
        flow_run_id = context["flow_run_id"]
        task_run_count = context.get("task_run_count", 1)
        idempotency_key = f"{flow_run_id}-{task_run_count}"
        flow_run_id = client.create_flow_run(
            flow_id=flow_id,
            parameters=parameters,
            idempotency_key=idempotency_key,
        )
```

Reply:
I have pushback on this - we shouldn't confuse this task raising an error with the Flow Run it created requesting a Flow-level retry, which isn't a well defined concept (the suggested implementation would create a new flow run for every "retry" event which isn't really a retry IMO).  _Any_ runtime error (including the task becoming a zombie) would result in an entirely new flow run.  I just can't imagine a scenario where that's desirable behavior.

Reply:
Sorry, I think we might be talking past each other. There are two types of task failures here:

- The flow run fails to submit. This is an error in this task.
- The flow run completes with a `Failed` state, a `FAILED` signal gets raised, and this task gets marked as failed.

If I'm using this task and set a retry on it, what is the retry for? Currently the retry only works for failures in submission - if the flow run fails the task will retry, but won't resubmit the flow run and I'll loop again in the retry loop.

Reply:
Perhaps this shouldn't set the success of this task as the success of the flow run, and instead should return the state of the flow run? Then a failure is always "failed to submit" rather than the flow run failing.

Reply:
> If I'm using this task and set a retry on it, what is the retry for
The retry is for robustness in the face of zombies - if the process dies + has retries, Cloud will resurrect it so the larger Flow Run can continue and this task run still still reflect the state of the submitted flow run.

Reply:
Also FYI The FAILED signal is a short circuit - it will bypass retry logic.

Reply:
> Also FYI The FAILED signal is a short circuit - it will bypass retry logic.

Ah, didn't realize this. Ok, that avoids the issue of endless retries. There's still not a good way to retry a failed flow run, but it sounds like that's not in-scope for this task. 👍 from me.